src/engine.rs

```diff
@@ -108,25 +108,31 @@ where
     let mut stream_to_start = None;
 
     let mixer = {
-        let mut end_points = engine.end_points.lock().unwrap();
-
-        match end_points.entry(device.name().expect("No device name")) {
-            Entry::Vacant(e) => {
-                let (mixer, stream) = new_output_stream(engine, device);
-                e.insert(Arc::downgrade(&mixer));
-                stream_to_start = Some(stream);
-                mixer
-            },
-            Entry::Occupied(mut e) => {
-                if let Some(m) = e.get().upgrade() {
-                    m.clone()
-                } else {
+        if let Ok(device_name) = device.name() {
```

Comment:
Can you remove the { around the if let? They aren't neccessary any more and save us an indentation level.

Reply:
you're right, totally missed that!